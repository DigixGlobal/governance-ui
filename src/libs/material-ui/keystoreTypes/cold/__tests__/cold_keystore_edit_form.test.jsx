import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ColdKeystoreEditForm from '../cold_keystore_edit_form';

const data = {
  type: 'testType',
  addresses: [{
    name: 'test address 1',
    address: '0x12345',
    tokens: [{
      id: 1,
    }, {
      id: 2,
    }],
    networks: [{
      id: 1,
    }, {
      id: 2,
    }],
  }],
};

describe('<ColdKeystoreEditForm />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <ColdKeystoreEditForm
        data={data}
        setFormData={jest.fn}
        formData={{}}
        formChange={jest.fn}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
