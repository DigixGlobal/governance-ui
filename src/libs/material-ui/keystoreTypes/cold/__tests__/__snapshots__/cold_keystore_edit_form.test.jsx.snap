// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColdKeystoreEditForm /> renders correctly 1`] = `
<ColdKeystoreCreationForm
  data={
    Object {
      "addresses": Array [
        Object {
          "address": "0x12345",
          "name": "test address 1",
          "networks": Array [
            Object {
              "id": 1,
            },
            Object {
              "id": 2,
            },
          ],
          "tokens": Array [
            Object {
              "id": 1,
            },
            Object {
              "id": 2,
            },
          ],
        },
      ],
      "type": "testType",
    }
  }
  editing={true}
  formChange={[Function]}
  formData={Object {}}
  setFormData={[Function]}
/>
`;
